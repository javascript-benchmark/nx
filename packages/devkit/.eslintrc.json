{
  "extends": "../../.eslintrc",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "@typescript-eslint/explicit-module-boundary-types": ["error"],
        "@typescript-eslint/no-restricted-imports": [
          "error",
          {
            "paths": [
              "@nx/workspace",
              "@angular-devkit/core",
              "@angular-devkit/architect",
              "@angular-devkit/schematics"
            ],
            "patterns": [
              {
                "group": ["nx/**/*"],
                "message": "Use requireNx() from packages/devkit/nx.ts OR use a type import instead.",
                "allowTypeImports": true
              },
              {
                "group": ["@nx/devkit/**/*"],
                "message": "Use a relative import"
              }
            ]
          }
        ]
      }
    },
    {
      "files": ["*.spec.ts"],
      "rules": {
        "@typescript-eslint/no-restricted-imports": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["./package.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/nx-plugin-checks": "error"
      }
    }
  ]
}
