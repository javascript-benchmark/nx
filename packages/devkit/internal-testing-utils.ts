/* eslint-disable @typescript-eslint/no-restricted-imports */
export * from 'nx/internal-testing-utils/assert-valid-migrations';
export * from 'nx/internal-testing-utils/run-migration-against-this-workspace';
export * from 'nx/internal-testing-utils/with-environment';
