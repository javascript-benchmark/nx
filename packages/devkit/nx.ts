// After Nx v18, this can be removed and replaced with either:
// - import {} from 'nx/src/devkit-exports'
// - import {} from 'nx/src/devkit-internals'
export function requireNx(): typeof import('nx/src/devkit-exports') &
  Partial<typeof import('nx/src/devkit-internals')> {
  try {
    let result = { ...require('nx/src/devkit-exports') };
    try {
      result = {
        ...result,
        // Remove in Nx v18, devkit should not support Nx v16.0.2 at that point.
        ...require('nx/src/devkit-internals'),
      };
    } catch {}
    return result;
  } catch {
    // Remove in Nx V17, devkit should not support Nx < 16 at that point.
    return require('./nx-reexports-pre16');
  }
}
