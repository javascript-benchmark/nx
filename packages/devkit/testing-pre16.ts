/* eslint-disable @typescript-eslint/no-restricted-imports */
export {
  createTreeWithEmptyWorkspace,
  createTreeWithEmptyV1Workspace,
} from 'nx/src/generators/testing-utils/create-tree-with-empty-workspace';
export { createTree } from 'nx/src/generators/testing-utils/create-tree';
