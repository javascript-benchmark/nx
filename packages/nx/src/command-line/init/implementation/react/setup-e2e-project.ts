import { writeFileSync } from 'fs';
import {
  fileExists,
  readJsonFile,
  writeJsonFile,
} from '../../../../utils/fileutils';

export function setupE2eProject(appName: string) {
  const json = readJsonFile(`apps/${appName}-e2e/project.json`);
  json.targets.e2e = {
    executor: 'nx:run-commands',
    options: {
      commands: [`nx e2e-serve ${appName}-e2e`, `nx e2e-run ${appName}-e2e`],
    },
  };
  json.targets['e2e-run'] = {
    executor: '@nx/cypress:cypress',
    options: {
      cypressConfig: `apps/${appName}-e2e/cypress.json`,
      tsConfig: `apps/${appName}-e2e/tsconfig.e2e.json`,
      baseUrl: 'http://localhost:3000',
    },
  };
  json.targets['e2e-serve'] = {
    executor: 'nx:run-commands',
    options: {
      commands: [`nx serve ${appName}`],
      readyWhen: 'can now view',
    },
  };
  writeJsonFile(`apps/${appName}-e2e/project.json`, json);

  if (fileExists(`apps/${appName}-e2e/src/integration/app.spec.ts`)) {
    const integrationE2eTest = `
      describe('${appName}', () => {
        beforeEach(() => cy.visit('/'));
        it('should contain a body', () => {
          cy.get('body').should('exist');
        });
      });`;
    writeFileSync(
      `apps/${appName}-e2e/src/integration/app.spec.ts`,
      integrationE2eTest
    );
  }
}
