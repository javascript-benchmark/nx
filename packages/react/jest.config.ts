/* eslint-disable */
export default {
  transform: {
    '^.+\\.[tj]sx?$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  globals: {},
  displayName: 'react',
  testEnvironment: 'node',
  preset: '../../jest.preset.js',
};
