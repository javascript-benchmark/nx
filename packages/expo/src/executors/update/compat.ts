import { convertNxExecutor } from '@nx/devkit';

import buildExecutor from './update.impl';

export default convertNxExecutor(buildExecutor);
