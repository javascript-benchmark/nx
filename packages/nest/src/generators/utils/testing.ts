import { addProjectConfiguration, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

export function createTreeWithNestApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  addProjectConfiguration(tree, appName, {
    root: `apps/${appName}`,
    sourceRoot: `apps/${appName}/src`,
    projectType: 'application',
    targets: {},
  });
  return tree;
}
