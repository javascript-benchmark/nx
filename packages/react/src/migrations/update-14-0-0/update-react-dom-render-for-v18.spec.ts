import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import {
  addProjectConfiguration,
  readJson,
  stripIndents,
  writeJson,
} from '@nx/devkit';
import update from './update-react-dom-render-for-v18';

describe('React update for Nx 14', () => {
  it('should remove deprecated @testing-library/react package', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    writeJson(tree, 'package.json', {
      devDependencies: {
        '@testing-library/react': '0.0.0',
      },
    });

    await update(tree);

    expect(readJson(tree, 'package.json')).toEqual({
      devDependencies: {
        '@testing-library/react': '0.0.0',
      },
    });
  });

  it.each`
    ext
    ${'js'}
    ${'jsx'}
    ${'tsx'}
  `('should update react-dom render call if it exists', async ({ ext }) => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(
      tree,
      'example',
      {
        root: 'apps/example',
        sourceRoot: 'apps/example/src',
        projectType: 'application',
        targets: {
          build: {
            executor: '@nrwl/web:webpack',
            options: {
              main: `apps/example/src/main.${ext}`,
            },
          },
        },
      },
      true
    );
    tree.write(
      `apps/example/src/main.${ext}`,
      stripIndents`
      import { Strict } from 'react';
      import * as ReactDOM from 'react-dom';
      import App from './app/app';
      
      ReactDOM.render(
        <Strict>
          <App/>
        </Strict>,
        document.getElementById('root')
      );
    `
    );

    await update(tree);

    expect(
      tree.read(`apps/example/src/main.${ext}`).toString()
    ).toMatchSnapshot();
  });

  it('should skip update if main file does not contain react-dom', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(
      tree,
      'example',
      {
        root: 'apps/example',
        sourceRoot: 'apps/example/src',
        projectType: 'application',
        targets: {
          build: {
            executor: '@nrwl/web:webpack',
            options: {
              main: `apps/example/src/main.tsx`,
            },
          },
        },
      },
      true
    );
    tree.write(
      `apps/example/src/main.tsx`,
      stripIndents`
        import { AppRegistry } from 'react-native';
        import App from './app/App';

        AppRegistry.registerComponent('main', () => App);
      `
    );

    await update(tree);

    expect(tree.read(`apps/example/src/main.tsx`).toString())
      .toMatchInlineSnapshot(`
      "import { AppRegistry } from 'react-native';
      import App from './app/App';

      AppRegistry.registerComponent('main', () => App);
      "
    `);
  });
});
