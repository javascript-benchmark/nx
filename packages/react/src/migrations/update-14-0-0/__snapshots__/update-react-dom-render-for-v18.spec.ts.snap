// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React update for Nx 14 should update react-dom render call if it exists 1`] = `
"import { Strict } from 'react';
import * as ReactDOM from 'react-dom/client';
import App from './app/app';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <Strict>
    <App />
  </Strict>
);
"
`;

exports[`React update for Nx 14 should update react-dom render call if it exists 2`] = `
"import { Strict } from 'react';
import * as ReactDOM from 'react-dom/client';
import App from './app/app';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <Strict>
    <App />
  </Strict>
);
"
`;

exports[`React update for Nx 14 should update react-dom render call if it exists 3`] = `
"import { Strict } from 'react';
import * as ReactDOM from 'react-dom/client';
import App from './app/app';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <Strict>
    <App />
  </Strict>
);
"
`;
